File: Domain/Activity.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Activity
    {
        // following properties will form columns in table "Activity"
        public Guid Id { get; set; }
        public string Title { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public bool IsCancelled { get; set; } // host can cancel this activity
        public ICollection<ActivityAttendee> Attendees { get; set; } = new List<ActivityAttendee>(); // initialise "attendees" property in Activity
    }
}

File: client-app/src/app/models/activity.ts
import { Profile } from "./profile";

export interface Activity {
  id: string
  title: string
  date: Date | null
  description: string
  category: string
  city: string
  venue: string
  hostUsername: string
  isCancelled: boolean
  isGoing: boolean
  isHost: boolean
  host?: Profile
  attendees: Profile[]
}

export class Activity implements Activity {
  // populate all properties into activity
  constructor(init?: ActivityFormValues) {
    Object.assign(this, init)
  }
}

// use a class to auto convert the activity object from API using constructor
export class ActivityFormValues {
  // properties needed in the create activity form
  id?: string = undefined
  title: string = ''
  category: string = ''
  description: string = ''
  date: Date | null = null
  city: string = ''
  venue: string = ''

  constructor(activity?: ActivityFormValues) {
    if (activity) {
      // map to view model
      this.id = activity.id
      this.title = activity.title
      this.category = activity.category
      this.description = activity.description
      this.date = activity.date
      this.venue = activity.venue
      this.city = activity.city
    }
  }
}
